File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def GCD(a, b):
    if a % b == 0:
        return b
    else:
        return GCD(b, a % b)

def solution(n, m):
    n, m = max(n, m), min(n, m)
    return [GCD(n, m), n * m // GCD(n, m)]

File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cs
using System;
public class Solution {
    public int GCD(int a, int b)
    {
        if (a % b == 0)
            return b;
        else
            return GCD(b, a % b);
    }
    
    public int[] solution(int n, int m) {
        int a = Math.Max(n, m);
        int b = Math.Min(n, m);
        return new int[] { GCD(a, b), a * b / GCD(a, b) };
    }
}